community/modules/compute/schedmd-slurm-gcp-v6-nodeset/outputs.tf

```diff
@@ -54,4 +54,28 @@ output "nodeset" {
     condition     = !var.enable_placement || !var.dws_flex.enabled
     error_message = "Cannot use DWS Flex with `enable_placement`."
   }
+
+  precondition {
+    condition     = var.reservation_name == "" || var.future_reservation == ""
+    error_message = "Cannot use reservations and future reservations in the same nodeset"
+  }
+
+  precondition {
+    condition     = !var.enable_placement || var.future_reservation == ""
+    error_message = "Cannot use `enable_placement` with future reservations."
+  }
+
+  precondition {
+    condition     = var.future_reservation == "" || length(var.zones) == 0
+    error_message = <<-EOD
+      If a future reservation is specified, `var.zones` should be empty.
+    EOD
+  }
+
+  precondition {
+    condition     = var.future_reservation == "" || length(regexall("/zones/${var.zone}/", local.nodeset.future_reservation)) > 0
```

Comment:
nit. can be simplified: `local.fr_zone == local.zones[0]`

Reply:
Noted, I can't use local.zones directly since its a set and hence unordered. But since we already check `var.zones` to be empty then we can use compared against `var.zone`.